bin/snyk_tests.sh

```diff
@@ -22,13 +24,56 @@ source bin/common/check_returncode.sh
 # authenticate snyk CLI locally
 run_and_check_returncode "snyk auth $SNYK_AUTHENTICATION_TOKEN"
 
-# Iac run limited to 20 runs a month. TBD Discuss with Jon Velapondi.
-# echo "Scanning Infrastructure as code with Snyk for potential vulnerabilities..."
-# snyk iac test || true
-
-# scan orca website
+# Scan orca website
 echo "Scanning open-source code with Snyk for potential vulnerabilities..."
 cd website
 snyk test -d --severity-threshold=high || true
+cd -
+
+function run_snyk_tests() {
+  echo
+  echo "Running snyk test in $1"
+  echo
+
+  cd $1
+  # remove shared library from the requirement files since snyk cannot scan through it
+  grep -v '../../shared_libraries' requirements-dev.txt >> test-requirements-dev.txt
+  grep -v '../../shared_libraries' requirements.txt >> test-requirements.txt
+  snyk test --severity-threshold=high --file=test-requirements-dev.txt --package-manager=pip || true
+  snyk test --severity-threshold=high --file=test-requirements.txt --package-manager=pip || true
+  # remove test files
+  rm test-requirements*.txt
+}
+export -f run_snyk_tests
+
+function run_snyk_tests_others() {
+  echo
+  echo "Running snyk test in $1"
+  echo
+  cd $1
+  # remove shared library from the requirement files since snyk cannot scan through it
+  grep -v '../shared_libraries' requirements-dev.txt >> test-requirements-dev.txt
+  grep -v '../shared_libraries' requirements.txt >> test-requirements.txt
+  snyk test --severity-threshold=high --file=test-requirements-dev.txt --package-manager=pip || true
+  snyk test --severity-threshold=high --file=test-requirements.txt --package-manager=pip || true
+  # remove test files
+  rm test-requirements*.txt
+}
+export -f run_snyk_tests_others
+
+task_dir_shared_lib=$(ls -d shared_libraries)
+task_dir_graphql=$(ls -d graphql)
+lambda_task_dirs=$(ls -d tasks/*)
+
+# Run snyk for shared library dir
+echo "Running snyk for shared library dir"
+parallel --jobs 0 -n 1 -X --halt now,fail=1 run_snyk_tests_others ::: $task_dir_shared_lib
+
+# Run snyk for graphql dir
+echo "Running snyk for graphql"
```

Comment:
Your function already has good logging, so I'd say you can remove this.

Reply:
removed